### PR TITLE
Introduce `apm-server.auth.*` config

### PR DESCRIPTION
**Pull request by axw**
## Motivation/summary

Introduce the new AgentAuth config structure, which holds API Key and secret token auth. Later we will add "anonymous" auth here too, and deprecate/replace some RUM config (rate limiting and allowed service names).

We also introduce a new YAML naming scheme for the config, `apm-server.auth.*`. The old config is deprecated and copied across to the new config fields.

## Checklist

- [x] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/master/CHANGELOG.asciidoc)
- [x] Documentation has been updated

## How to test these changes

1. Set `apm-server.secret_token` and `apm-server.api_key.*`, make sure they are honoured (e.g. query "GET /" with/out auth)
2. Set `apm-server.auth.secret_token` and `apm-server.auth.api_key.*`, same again.
3. Set both new and old with different values: check that a warning is logged that the old config is ignored, and check that it is possible to auth using the new but not old

## Related issues

https://github.com/elastic/apm-server/issues/5347

### CODE REVIEW

File: docs/configuration-process.asciidoc [hunk 1]
```diff
@@ -99,6 +103,8 @@ The token is not used for RUM endpoints. By default, no authorization token is s
 It is recommended to use an authorization token in combination with SSL enabled.
 Read more about <<securing-apm-server, Securing APM Server>> and the <<secret-token, secret token>>.
 
+deprecated::[7.14.0, "In versions prior to 7.14.0, secret token authorization was known as `apm-server.secret_token`. In 7.14.0 this was renamed `apm-server.auth.secret_Token`. The old configuration will continue to work until 8.0.0, and the new configuration will take precedence."]
```
Comment (simitt) [inline docs/configuration-process.asciidoc:106]: typo: `secret_Token`

**Review by simitt — COMMENTED** [review]
Looks great.

File: docs/configuration-process.asciidoc (hunk 1 above)
Comment (axw) [reply docs/configuration-process.asciidoc]: thanks, fixed

**Comment by axw** [comment]
@bmorelli25 would you please take a look at the docs changes? I've renamed some config, and added new deprecation sections for the deprecated config names.

I've also moved "api_key.* configuration options" into the "API keys" section, not sure if it was intentional that it was separate? I can move back if you prefer, but then I'm not sure where the deprecation section should go.

**Review by simitt — CHANGES_REQUESTED** [review]
Can you also change the `apmpackage` please. I suggest we remove the deprecated settings and only support the new ones.

**Comment by axw** [comment]
@simitt will do. I was going to wait for your changes (https://github.com/elastic/apm-server/pull/5444) to land, but I'll just update now for the purposes of review and update again when that lands.

**Review by simitt — APPROVED** [review]

**Review by bmorelli25 — APPROVED** [review]
Docs look great–thanks!

It looks like there's one [additional change](https://github.com/elastic/apm-server/blame/master/docs/copied-from-beats/docs/command-reference.asciidoc/#L142-L143) that needs to be made to the command reference file. I'll fix that in the Beats repo and copy it over to apm-server before 7.14.

**Comment by axw** [comment]
Failure is related to the beats update, which will be resolved by https://github.com/elastic/apm-server/pull/5471

**Comment by simitt** [comment]
Tested with BC2:
* old settings work as expected
* new settings work as expected
* API Key auth disabled when:
```
apm-server.api_key.enabled: true
apm-server.auth.api_key.enabled: false
```
  * Secret token 'abcd' expected
```
apm-server.secret_token: 'xxx'
apm-server.auth.secret_token: 'abcd'
```

**Pull request merged**